packages/workbox-background-sync/src/Queue.ts

```diff
@@ -361,7 +382,8 @@ class Queue {
 
             let syncError;
             try {
-              await this._onSync({queue: this});
+              // TODO (philipwalton): Should the await be here?
```

Comment:
@philipwalton

Reply:
Yes, we have to await here. If we don't—and if the `onSync` function is an async function (which is should be)—then any errors thrown during the execution of that function won't be caught.

We could type that `onSync` returns a promise, but technically it might not since it's defined by the user.

Reply:
Okay, I'll change

```
interface OnSyncCallback {
  (options: OnSyncCallbackOptions): void;
}
```

to

```
interface OnSyncCallback {
  (options: OnSyncCallbackOptions): Promise<void>;
}
```

in that case. 

(Again, really not sure if this needs to be marked as a breaking change?)